internal/flex/write_only.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package flex
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+)
+
+type writeOnlyAttrGetter interface {
+	GetRawConfigAt(path cty.Path) (cty.Value, diag.Diagnostics)
+	Id() string
+}
+
+// GetWriteOnlyValue returns the value of the write-only attribute from the config.
+func GetWriteOnlyValue(d writeOnlyAttrGetter, path cty.Path, attrType cty.Type) (cty.Value, diag.Diagnostics) {
```

Comment:
Is there any value to adding typed wrappers on top of this, e.g. `GetWriteOnlyStringValue` that returns a `string`?

Reply:
@ewbankkit worth adding and should eliminate some complex checks. We can start with string then add as we see more types